unison-runtime/src/Unison/Runtime/Decompile.hs

```diff
@@ -222,7 +217,7 @@ decompileForeign backref topTerms f
         _ -> l
   | Just l <- maybeUnwrapForeign typeLinkRef f =
       pure $ typeLink () l
-  | Just (a :: Array Closure) <- maybeUnwrapForeign iarrayRef f =
+  | Just (a :: Array Val) <- maybeUnwrapForeign iarrayRef f =
```

Comment:
I missed one or two of these that resulted in annoying-to-find segfaults, is there a better way we could prevent that?

Granted, these sorts of sweeping change PRs will hopefully be pretty rare, so I'm happy if the answer is no, or it's not worth it.

Reply:
I could probably come up with a different way of accomplishing what's going on there that would be more reliable when you change things. But you'd still have to find all the places where we do that right now.